homeassistant/components/sensor/iperf3.py

```diff
@@ -0,0 +1,230 @@
+"""
+Support for Iperf3 network measurement tool.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.iperf3/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN, PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_MONITORED_CONDITIONS, STATE_UNKNOWN)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import slugify
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_change
+from homeassistant.helpers.restore_state import async_get_last_state
+import homeassistant.util.dt as dt_util
+
+REQUIREMENTS = ['iperf3==0.1.10']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_PROTOCOL = 'Protocol'
+ATTR_REMOTE_HOST = 'Remote Server'
+ATTR_REMOTE_PORT = 'Remote Port'
+ATTR_TEST_STATUS = 'Test Status'
+ATTR_VERSION = 'Version'
+
+CONF_ATTRIBUTION = 'Data retrieved using Iperf3'
+CONF_SECOND = 'second'
+CONF_MINUTE = 'minute'
+CONF_HOUR = 'hour'
+CONF_DAY = 'day'
+CONF_MANUAL = 'manual'
+CONF_DURATION = 'duration'
+CONF_SERVER = 'server'
+CONF_PORT = 'port'
+
+DEFAULT_DURATION = 10
+DEFAULT_PORT = 5201
+
+ICON = 'mdi:speedometer'
+
+SENSOR_TYPES = {
+    'download': ['Download', 'Mbit/s'],
+    'upload': ['Upload', 'Mbit/s'],
+}
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(list(SENSOR_TYPES))]),
+    vol.Required(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_SERVER): cv.string,
+    vol.Optional(CONF_DURATION, default=DEFAULT_DURATION): vol.Range(5, 10),
+    vol.Optional(CONF_SECOND, default=[0]):
+        vol.All(cv.ensure_list, [vol.All(vol.Coerce(int), vol.Range(0, 59))]),
+    vol.Optional(CONF_MINUTE, default=[0]):
+        vol.All(cv.ensure_list, [vol.All(vol.Coerce(int), vol.Range(0, 59))]),
+    vol.Optional(CONF_HOUR):
+        vol.All(cv.ensure_list, [vol.All(vol.Coerce(int), vol.Range(0, 23))]),
+    vol.Optional(CONF_DAY):
+        vol.All(cv.ensure_list, [vol.All(vol.Coerce(int), vol.Range(1, 31))]),
+    vol.Optional(CONF_MANUAL, default=False): cv.boolean,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Iperf3 sensor."""
+    data = Iperf3Data(hass, config)
+
+    dev = []
+    for sensor in config[CONF_MONITORED_CONDITIONS]:
+        dev.append(Iperf3Sensor(data, sensor))
+
+    add_devices(dev)
+
+    def update(call=None):
+        """Update service for manual updates."""
+        data.update(dt_util.now())
+        for sensor in dev:
+            sensor.update()
+
+    for sensor in dev:
+        hass.services.register(DOMAIN, sensor.service_name, update)
```

Comment:
All services need to be always prefixed with the platform name.

Reply:
@balloob the service name is prefixed with the platform name as I'm defined here:

```
  94     @property
  95     def service_name(self):
  96         """Return the service name of the sensor."""
  97         return slugify("{} {}".format(
  98             'update_iperf3', self.iperf3_client.server))
```

This allows the user to have more than 1 `iperf3` component set and run it manually as a service, right? If I just set this to 'update_iperf3', only the last sensor on the `dev` list will make the list of services. Is there a way to keep only 1 service name but to trigger actions on different sensors?